File: test_dir/api/redis_insert.py
#! /usr/bin/python3
# -*- coding: utf-8 -*-
#
#	test_dir/api/redis_insert.py
#
#					Jan/14/2020
#
# --------------------------------------------------------
import	sys
import	os
import	string
import  json
import  redis
import  cgi
import	cgitb
cgitb.enable()
#
#
# --------------------------------------------------------
def get_string_proc(ff,key):
    if key in ff:
        value = ff.getfirst(key,"")
    else:
        value = "0000"
#
    return value
# --------------------------------------------------------
sys.stderr.write("*** redis_insert.py *** start *** \n")
#
ff=cgi.FieldStorage()
#
key = get_string_proc(ff,"key")
json_str = get_string_proc(ff,"value")
#
if json_str != "0000":
	rr = redis.Redis(host='localhost', port=6379, db=0)
	rr.set(key, json_str)
#
print ("Content-type: text/json; charset=UTF-8\n\n")
print (json_str)

sys.stderr.write("*** redis_insert.py *** end *** \n")
#
# --------------------------------------------------------

File: test_dir/ruby/api/redis_insert.rb
#! /usr/bin/ruby
# -*- coding: utf-8 -*-
# ---------------------------------------------------------------------
#	redis_insert.ruby
#
#					Jan/16/2020
#
# ---------------------------------------------------------------------
require 'redis'
require	"cgi"
require	"json"
#
STDERR.puts	"*** 開始 ***"
#
$cgi=CGI.new
key = $cgi["key"]
value = $cgi["value"]

redis = Redis.new(:host => "localhost", :port => 6379)
redis.set key,value

puts "Content-type: text/json; charset=UTF-8\n\n"
#
puts value
#
STDERR.puts	"*** 終了 ***"
